File: examples/sharding/easy_sqs.py
import boto3
import json

class EasySqs():

    def __init__(self, config):
        self.sqs_client = boto3.client('sqs', 
            aws_access_key_id=config["aws_access_key_id"], 
            aws_secret_access_key=config["aws_secret_access_key"], 
            region_name=config["region_name"])

    def send_message(self, queue_url, message):
        # Send message to SQS queue
        response = self.sqs_client.send_message(
            QueueUrl=queue_url,
            DelaySeconds=10,
            MessageBody=(
                message
            )
        )
        return response

    def receive_message(self, queue_url):
        response = self.sqs_client.receive_message(
            QueueUrl=queue_url,
            MessageAttributeNames=[
                'string',
            ],
            MaxNumberOfMessages=1,
            VisibilityTimeout=10,
            WaitTimeSeconds=10
        )
        messages = response["Messages"]
        message = messages[0]
        return message

    def delete_message(self, queue_url, receipt_handle):
        response = self.sqs_client.delete_message(
            QueueUrl=queue_url,
            ReceiptHandle=receipt_handle
        )
        return response

File: examples/sharding/frontend/js/EasySQS.js
// const AWS = require('aws-sdk');
var sqs;

class EasySQS {
    constructor(config) {
        AWS.config.update(config);
        AWS.config.update({region: 'us-east-1'});
        sqs = new AWS.SQS();
    }

    send(message, queueUrl) {
    	var params = {
    		MessageBody: message,
    		QueueUrl: queueUrl,
    		DelaySeconds: 0
    	};

    	
    	return new Promise((resolve, reject) => {
	    	sqs.sendMessage(params, (err, data) => {
	    		if (err) {
	    			reject(err);
	    		} else {
	    			resolve(data);
	    		}
	    	});
	    });
    }

    receive(queueUrl) {
    	var params = {
    		QueueUrl: queueUrl,
    		VisibilityTimeout: 100 // 10 second wait time
    	};

    	
    	return new Promise((resolve, reject) => {
	    	sqs.receiveMessage(params, (err, data) => {
	    		if (err) {
	    			reject(err);
	    		} else {
	    			resolve(data);
	    		}
	    	});
	    });
    }

    delete(receiptHandle, queueUrl) {
    	var params = {
    		QueueUrl: queueUrl,
    		ReceiptHandle: receiptHandle
    	};
    	
    	return new Promise((resolve, reject) => {
	    	sqs.deleteMessage(params, (err, data) => {
	    		if (err) {
	    			reject(err);
	    		} else {
	    			resolve(data);
	    		}
	    	});
	    });
    }
}

export {EasySQS};